utils/log.go

```diff
@@ -0,0 +1,83 @@
+package utils
+
+import (
+	"errors"
+
+	"github.com/dgraph-io/badger/v3"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+var ErrUnknownLogLevel = errors.New("unknown log level")
+
+type LogLevel uint8
+
+const (
+	DEBUG LogLevel = iota
+	INFO
+	WARN
+	ERROR
+)
+
+func (l LogLevel) String() string {
+	switch l {
+	case DEBUG:
+		return "debug"
+	case INFO:
+		return "info"
+	case WARN:
+		return "warn"
+	case ERROR:
+		return "error"
+	default:
+		return ""
+	}
+}
+
+func (l LogLevel) IsValid() bool {
+	return l.String() != ""
+}
+
+type Logger interface {
+	SimpleLogger
+	badger.Logger
+}
+
+type SimpleLogger interface {
+	Debugw(msg string, keysAndValues ...any)
+	Infow(msg string, keysAndValues ...any)
+	Warnw(msg string, keysAndValues ...any)
+	Errorw(msg string, keysAndValues ...any)
```

Comment:
Are these the functions implemented by `zap.SugaredLogger`? Is that why they all end with `w`?

Reply:
Yes, it implements those methods. I think the `w` stands for "with" (the message is printed *with* the keys and values).